src/defineFunction.js

```diff
@@ -128,27 +129,44 @@ export type FunctionSpec = {|
     numOptionalArgs: number,
     infix: boolean,
     consumeMode: ?Mode,
+
+    // FLOW TYPE NOTES: Doing either one of the following two
+    //
+    // - removing the NOTETYPE type parameter in FunctionSpec above;
+    // - using ?FunctionHandler<NODETYPE> below;
```

Comment:
I would've expected `?FunctionHandler<NODETYPE>` to work. :\

Reply:
Me too! This error is very confusing. ;(
I didn't yet try to do a minimally-reproduceable example to file a bug on flow.

Reply:
Well, at least we have a detailed comment here that we can dig into later.